extmod/moductypes.c

```diff
@@ -602,7 +602,7 @@ static mp_obj_t uctypes_struct_unary_op(mp_unary_op_t op, mp_obj_t self_in) {
                 uint agg_type = GET_TYPE(offset, AGG_TYPE_BITS);
                 if (agg_type == PTR) {
                     byte *p = *(void **)self->addr;
-                    return mp_obj_new_int((mp_int_t)(uintptr_t)p);
+                    return mp_obj_new_int_from_uint((mp_int_t)(uintptr_t)p);
```

Comment:
```suggestion
                    return mp_obj_new_int_from_uint((uintptr_t)p);
```

Probably don't need the 2nd cast since the parameter type is `mp_uint_t` already.

Reply:
Heh, I'm too slow, Angus beat me to it.

Reply:
Two out of two reviewers agree! :grin: